src/fileformats/ocd_file_export.h

```diff
@@ -249,11 +249,13 @@ class OcdFileExport : public Exporter
 	        const LineSymbol* double_line );
 	
 	
-	void exportSymbolIcon(const Symbol* symbol, Ocd::IconV8& icon);
+public:
+	static void exportSymbolIcon(const Map* map, const Symbol* symbol, Ocd::IconV8& icon);
 	
-	void exportSymbolIcon(const Symbol* symbol, Ocd::IconV9& icon);
+	static void exportSymbolIcon(const Map* map, const Symbol* symbol, Ocd::IconV9& icon);
```

Comment:
Think so too.

Reply:
Returning by value instead of passing an output parameter makes it impossible to simply overload the function name for different icon types. We will either have to use templates, or we may use an extra type, moving the export implementation to conversion operators:

~~~
struct OcdIcon
{
	const Map& map;
	const Symbol& symbol;
	
	operator Ocd::IconV8() const;
	operator Ocd::IconV9() const;
};

...

ocd_base_symbol.icon = OcdIcon{map, symbol};
~~~

I admit there is some beauty in this approach: It eliminates static functions from OcdFileExport, and `OcdIcon` totally encapsulates functionality which might be used and tested separately.

Reply:
OcdIcon is key contribution of #1136.